dht_net.go

```diff
@@ -207,12 +209,38 @@ func (dht *IpfsDHT) handleNewMessage(s network.Stream) bool {
 	}
 }
 
-// sendRequest sends out a request, but also makes sure to
+type messageManager struct {
```

Comment:
can this move to its own file?

Reply:
Sure. Question though, should we export this? It seems like it's just a generic thing that sends varint-prefixed protobufs over libp2p with some semantics around trying to reuse streams.

We may also want to use this in the crawler, but if the crawler is the only user then we could just move this to the `internal` package.